### PR TITLE
Anchor output support (iff EXPERIMENTAL_FEATURES)

### PR DESCRIPTION
**Pull request by rustyrussell**
Based on the trivial #3829, this implements anchor outputs (https://github.com/lightningnetwork/lightning-rfc/pull/688).  It passes lnprototest, but still needs testing against lnd.

Changelog-None, since it's EXPERIMENTAL_FEATURES only.

### CODE REVIEW

File: tests/test_plugin.py [hunk 1]
```diff
@@ -1396,6 +1396,26 @@ def test_coin_movement_notices(node_factory, bitcoind, chainparams):
             {'type': 'chain_mvt', 'credit': 100001000, 'debit': 0, 'tag': 'deposit'},
             {'type': 'chain_mvt', 'credit': 941045000, 'debit': 0, 'tag': 'deposit'},
         ]
+    elif EXPERIMENTAL_FEATURES:
+        # option_anchor_outputs
+        l2_l3_mvts = [
+            {'type': 'chain_mvt', 'credit': 1000000000, 'debit': 0, 'tag': 'deposit'},
+            {'type': 'channel_mvt', 'credit': 0, 'debit': 100000000, 'tag': 'routed'},
+            {'type': 'channel_mvt', 'credit': 50000501, 'debit': 0, 'tag': 'routed'},
+            {'type': 'chain_mvt', 'credit': 0, 'debit': 8430501, 'tag': 'chain_fees'},
```
Comment (niftynei) [inline tests/test_plugin.py:1405]: 8430.501 - 5430.501 =  3000.000 sats. definitely more expensive!

File: hsmd/hsmd.c [hunk 2]
```diff
@@ -1056,8 +1058,16 @@ static struct io_plan *handle_sign_remote_htlc_tx(struct io_conn *conn,
 		return bad_req_fmt(conn, c, msg_in,
 				   "Failed deriving htlc pubkey");
 
+	/* BOLT-a12da24dd0102c170365124782b46d9710950ac1 #3:
+	 * ## HTLC-Timeout and HTLC-Success Transactions
+	 *...
+	 * * if `option_anchor_outputs` applies to this commitment transaction,
+	 *   `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY` is used.
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply hsmd/hsmd.c:1065]: Yes, it's still in.

File: onchaind/onchaind.c [hunk 3]
```diff
@@ -732,7 +732,7 @@ new_tracked_output(struct tracked_output ***outs,
 		   enum output_type output_type,
 		   const struct htlc_stub *htlc,
 		   const u8 *wscript,
-		   const secp256k1_ecdsa_signature *remote_htlc_sig)
+		   const struct bitcoin_signature *remote_htlc_sig TAKES)
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply onchaind/onchaind.c:738]: tal_dup does that for us...

File: openingd/openingd.c [hunk 4]
```diff
@@ -243,6 +244,18 @@ static bool check_config_bounds(struct state *state,
 		return false;
 	}
 
+	/* If they're opener, they must pay for anchor outputs, so include
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply openingd/openingd.c:247]: Let me include BOLT quote...

File: bitcoin/psbt.c [hunk 5]
```diff
@@ -464,6 +464,44 @@ struct wally_tx *psbt_finalize(struct wally_psbt *psbt, bool finalize_in_place)
...
+		 * #### `to_remote` Output
+		 *...
+		 *
+		 * If `option_anchor_outputs` applies to the commitment
+		 * transaction, the `to_remote` output is encumbered by a one
+		 * block csv lock.
+		 *
+		 *    <remote_pubkey> OP_CHECKSIGVERIFY 1 OP_CHECKSEQUENCEVERIFY
+		 *
+		 * The output is spent by a transaction with `nSequence` field set to `1` and witness:
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply bitcoin/psbt.c:491]: Will be fixed automatically when we update BOLT (which has that text)...

**Comment by rustyrussell** [comment]
Rebased (db change conflicts), and added BOLT comment as per @niftynei hint.

**Comment by rustyrussell** [comment]
Fairly simple rebase, fixing up a missing db select addition for new field in previous rebase...

**Comment by rustyrussell** [comment]
Trivial rebase to fix test_plugin conflict.

**Pull request merged**